### PR TITLE
extend typing

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/metric.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 from torchmetrics.utilities.imports import _LIGHTNING_AVAILABLE, _compare_version
 
 
-def distributed_available() -> bool:
+def jit_distributed_available() -> bool:
```
Comment (maximsch2) [inline torchmetrics/metric.py:33]: jit sounds like something related to TorchScript, but I see what you mean and it makes sense in a literal sense of it.

File: torchmetrics/metric.py [hunk 2]
```diff
@@ -359,7 +359,7 @@ def reset(self):
             else:
                 setattr(self, attr, [])
 
-    def clone(self):
+    def clone(self) -> "Metric":
```
Comment (maximsch2) [inline torchmetrics/metric.py:362]: if you do `from __future__ import annotations` at the top then you can drop the quotes.

File: torchmetrics/metric.py [hunk 3]
```diff
@@ -330,19 +330,19 @@ def wrapped_func(*args, **kwargs):
...
     @abstractmethod
-    def update(self) -> None:  # pylint: disable=E0202
+    def update(self, *_: Any, **__: Any) -> None:  # pylint: disable=E0202
         """
         Override this method to update the state variables of your metric class.
         """
 
     @abstractmethod
-    def compute(self):  # pylint: disable=E0202
+    def compute(self) -> Any:  # pylint: disable=E0202
```
Comment (maximsch2) [inline torchmetrics/metric.py:339]: IIRC TorchScript will not understand that now, but probably the right thing here is to define a type parameter T and have Metric[T] and then `compute() -> T`

**Review by maximsch2 — APPROVED** [review]

File: torchmetrics/metric.py (hunk 2 above)
Comment (Borda) [reply torchmetrics/metric.py:362]: lets leave it to another PR as it will be in multiple places

File: torchmetrics/metric.py (hunk 3 above)
Comment (Borda) [reply torchmetrics/metric.py:339]: ok, i ll drop it for now, but mind send a PR?

**Pull request merged**